### PR TITLE
Fixes several null reference exceptions.

### PR DESCRIPTION
**Pull request by mkrueger**
Fixes issue #5666 Exception in RegexCompletionProvider 
Fixes issue #5665 NullRef in HighlightUrlExtension
Fixes issue #5660 NullRef in Roslyn

### CODE REVIEW

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Rename/RenameRefactoring.cs [hunk 1]
```diff
@@ -82,6 +82,16 @@ static void Rollback (TextEditor editor, List<MonoDevelop.Core.Text.TextChangeEv
 		public async Task Rename (ISymbol symbol)
 		{
 			var ws = IdeApp.Workbench.ActiveDocument.RoslynWorkspace;
+			if (!symbol.IsDefinedInSource ())
+				return;
+			foreach (var location in symbol.Locations)
+			{
+				if (location.IsInSource && ws.CurrentSolution.GetDocument (location.SourceTree) == null) {
+					LoggingService.LogError ("Error location.SourceTree document not found.");
+					MessageService.ShowError ("Can't rename " + symbol.Name + ". If a retry doesn't work please file a bug.");
```
Comment (DavidKarlas) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Rename/RenameRefactoring.cs]: This probably needs to be localizable.

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/RegexCompletionProvider.cs [hunk 2]
```diff
@@ -113,6 +113,8 @@ public override async Task ProvideCompletionsAsync (Microsoft.CodeAnalysis.Compl
 						}
 					}
 				}
+			} catch (TaskCanceledException) {
```
Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Completion/CompletionProvider/RegexCompletionProvider.cs]: `OperationCanceledException` here.

File: main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Rename/RenameRefactoring.cs [hunk 3]
```diff
@@ -181,9 +191,8 @@ public class RenameProperties
 		public async Task<List<Change>> PerformChangesAsync (ISymbol symbol, RenameProperties properties)
 		{
 			var ws = IdeApp.Workbench.ActiveDocument.RoslynWorkspace;
-
-			var newSolution = await Renamer.RenameSymbolAsync (ws.CurrentSolution, symbol, properties.NewName, ws.Options);
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Rename/RenameRefactoring.cs]: why move this?

Comment (mkrueger) [reply main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Rename/RenameRefactoring.cs]: removed ws and moved 'changes' - that's what the diff makes about it - but I moved changes now more to the bottom.

**Comment by slluis** [comment]
@monojenkins backport release-7.6

**Pull request merged**